lib/resources/file.rb

```diff
@@ -109,16 +124,14 @@ def to_s
 
     private
 
-    def file_permission_granted?(flag, by_usergroup, by_specific_user)
-      unless inspec.os.unix?
-        fail 'Checking file permissions is not supported on your os'
-      end
-
+    def file_permission_granted?(access, by_usergroup, by_specific_user)
+      return nil if @perms_provider.nil?
```

Comment:
we should fail here, once we have the skip as the first statement of eg. in `readable?` this method should never be called if perms_provider is not defined.